### PR TITLE
Run in Section for notebook sticky scroll context menu

### PR DESCRIPTION
**Pull request by Yoyokrazy**

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 1]
```diff
@@ -56,6 +63,43 @@ export class ToggleNotebookStickyScroll extends Action2 {
...
+		const isNotebook = isNotebookEditorInput(activeInput);
+		if (!isNotebook) {
+			return;
+		}
+		const notebookEditor = notebookEditorService.retrieveExistingWidgetFromURI(activeInput.resource)?.value;
+		if (!notebookEditor) {
+			return;
+		}
+
+		const notebookStickyScrollComponent = notebookEditor.notebookStickyScroll;
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:98]: maybe `notebookStickyScrollComponent` can also come from args.

**Comment by rebornix** [comment]
@Yoyokrazy do you get a chance to update the PR for further review?

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 2]
```diff
@@ -208,6 +245,18 @@ export class NotebookStickyScroll extends Disposable {
...
+			actionRunner: {
+				run: async (action) => {
+					if (action.id === 'notebook.action.runInSection') {
+						this.commandService.executeCommand(action.id, event.target, this.currentStickyLines, this.notebookEditor);
+					} else {
+						this.commandService.executeCommand(action.id);
+					}
+				},
+				onDidRun: new Emitter<IRunEvent>().event,
+				onWillRun: new Emitter<IRunEvent>().event,
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:257]: from what I read https://github.com/microsoft/vscode/blob/673a1aeca1ed358222c25de98570f23060997b49/src/vs/platform/contextview/browser/contextMenuHandler.ts#L85-L86 , the context menu dismisses when run* events are triggered so we can't just have a dummy event emitter here.

Comment (Yoyokrazy) [reply src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:257]: I'll go ahead and implement a more thorough actionRunner then 👍

**Review by rebornix — COMMENTED** [review]
Looks good to me in overall. We can merge this in debt week once the context menu issue is resolved.

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 3]
```diff
@@ -205,9 +245,17 @@ export class NotebookStickyScroll extends Disposable {
...
+		const context: RunInSectionContext = {
+			target: event.target,
+			currentStickyLines: this.currentStickyLines,
+			notebookEditor: this.notebookEditor,
+		};
+
 		this._contextMenuService.showContextMenu({
 			menuId: MenuId.NotebookStickyScrollContext,
 			getAnchor: () => event,
+			menuActionOptions: { shouldForwardArgs: true, arg: context },
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:258]: 👍 , I forgot about this option

File: src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts [hunk 4]
```diff
@@ -395,6 +444,7 @@ export class NotebookStickyScroll extends Disposable {
 
 	override dispose() {
 		this._disposables.dispose();
+		// this._actionRunnerDisposables.dispose();
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts:447]: nit, remove the commented code.

**Review by rebornix — APPROVED** [review]

**Pull request merged**